Only evaluate the first template definition

ternite opened the pull request:
Uses the [usecounter](https://www.dokuwiki.org/plugin:usecounter) plugin to be able to restrict multiple template definitions to the first one, instead of the last one.

Comes with a boolean configuration option `firsttemplatedefinitionwins`. If set to true, only the first template command will be evaluated. If set to false, the original behavior will be enacted, i.e. all commands are executing, which results in the last command actually setting the template.

resolves #281

Klap-in commented:
For the settings via syntax I did saw two other templates `odt_template` or `css_template` settings, but not yet `template`. Are these also supported?
```
{{odt>setting:value}}
```
https://www.dokuwiki.org/plugin:odt#configuration

Further, it seems you are using tabs. Could you please change these in spaces? 

I understand this change as not completely backward compatible. So in the changelog we need to pay attention to it.

ternite commented:
@Klap-in Thanks for your comment! I wasn't aware of the other configuration options. Looks like I implemented my wiki when those were not existing, yet.
I made an update to the branch with the following properties:
- The syntax tags `css_template`, `odt_template`, and `templatepage` will be counted as a usage of template, too.
- Added more commenting and documentation - the most important detail is mentioning in the plugin configuration text for `firsttemplatedefinitionwins` that this option needs the [usecounter](https://www.dokuwiki.org/plugin:usecounter) plugin to work.
- Changed default value to 0, so plugin behavior will be the same as in previous versions of the plugin, even if the usecounter plugin is installed and activated. _(Does this resolve your considerations towards backward compatibility?)_
- Changed the occurrences of tab characters to spaces.

Here are the additions that could be made to the dokuwiki page of the [odt](https://www.dokuwiki.org/plugin:odt) plugin:

In section "Configuration" - addition of:
```
| firsttemplatedefinitionwins      | In case of multiple definitions of template, templatepage, css_template, or odt_template, the first will be used. (The value false will apply the last definition, instead.) **Needs the usecounter plugin!**                                                                                              |
```

In section "Templates support" - addition of a new paragraph right before the next section "PDF conversion":
```
=== Use the first or last template defition? ===

In a setting with nested pages, e.g. when using the [[plugin:include|include plugin]], the classic behavior of this plugin was to use the very last template definition, when multiples were made. This may be undesired, so there is a configuration setting ''firsttemplatedefinitionwins'' which changes this behavior to evaluating only the very first template definition. This configuration parameter needs the [[plugin:usecounter]] plugin to work! Also, the default value of this config option is ''0'' (''false''), so it does not change the plugin behavior if you upgraded from an earlier version of this plugin.
```

Best regards
Thomas

Klap-in commented:
Thanks for improving. 
I did not realise the dependency of the usecounter plugin, even while you mentioned it already. My feeling is that implementing a counter directly in odt can be quite straight forward, this extra plugin does it nice, but for me it still feels as extra burden for the user to install an extra plugin only for just this setting and it is an extra possibility for confusion. Would it not simplify the usage of the setting to implement the counter directly in odt plugin?

With the new default setting the situation is indeed backward compatible.

Nice that you propose already the documentation updates.

ternite commented:
I implemented it this way because I am using another private plugin which needs the same functionality. Also, since you have to actively opt-in the new behavior, you will have to manually set the configuration value and you're reminded therein of this dependency. I don't feel it's much of a burden to install another small plugin if you want/need this functionality.

Also, there is way more code in that plugin than in this branch, here. You need an action handler, and a helper class. To integrate that code into plugin:odt seemed complicating the matter a lot in relativity to the few lines added within this branch. I did not want to blow up the odt plugin even more than it already is, just because of some niche functionality (I seem to be the first person to perceive the resolved problem).

Klap-in commented on helper/config.php:
```diff
@@ -184,6 +184,15 @@ public function __construct() {
+        // firsttemplatedefinitionwins
+        $this->config ['firsttemplatedefinitionwins'] =
+            array('value'              => NULL,
+                  'DWGlobalName'       => 'firsttemplatedefinitionwins',
```
DokuWiki has not a setting with this name, so mentioning this here will have no use?

ternite replied:
Ouch, I guess you're right. I was mislead by the name `$this->config` and believed at the time of writing that the plugin configuration setting was made in a dynamic way, there. Instead, it looks like it's setting up the possible tags that can be used in the page syntax. Since my changes do not add any tags, the changes in helper/config.php can be dropped, obviously.

ternite commented:
@Klap-in hi, you merged master into this branch - will you be going to merge the branch into master, too?

Klap-in commented on helper/config.php:
```diff
@@ -184,6 +184,15 @@ public function __construct() {
+        // firsttemplatedefinitionwins
+        $this->config ['firsttemplatedefinitionwins'] =
+            array('value'              => NULL,
+                  'DWGlobalName'       => 'firsttemplatedefinitionwins',
```
```suggestion
                  'DWGlobalName'       => null,
```
I propose to remove the name here.

Klap-in replied:
This global setting 'firsttemplatedefinitionwins' does not exist in DokuWiki's config. Therefore, to prevent confusing, I prefer let it search for it. Might be that the rest of this config setting has some other nuances, but it looks similar as the other settings, so I expect it will work fine.
